src/provider/github.rs

```diff
@@ -89,17 +91,22 @@ impl Provider for GitHub {
         let mut mirrors: Vec<MirrorResult> = Vec::new();
 
         for p in projects {
+            let mut branch = "";
             match serde_yaml::from_str::<Desc>(&p.description.unwrap_or_default()) {
                 Ok(desc) => {
                     if desc.skip {
                         mirrors.push(Err(MirrorError::Skip(p.url)));
                         continue;
                     }
+                    if !desc.branch.is_empty() {
+                        branch = &desc.branch
+                    }
```

Comment:
Why this extra step? Just use `desc.branch` directly, like for `origin`.

To make it optional I think it is better to use an `Option<String>` so the intention is clear. I think `serde` should be able to handle that.

Reply:
good point.